### PR TITLE
Add hover activity over Study code in the Participants Table

### LINKED ISSUE #670: Add hover activity over Study code in the Participants Table
People have a hard time figuring out which short study names map to the long names.

This feature was requested by Marty (PI).

Image

### PR DESCRIPTION
**Pull request by dheshanm**
Closes #670

### CODE REVIEW

**Review by JackVCurtis — APPROVED** [review]
Looks good!

File: views/tables/ParticipantsTable/index.jsx [hunk 1]
```diff
@@ -135,14 +136,21 @@ const ParticipantsTable = (props) => {
           />
         )
       case 'study':
+        let tooltip_title = participant[property]
+        if (participant[property] in SITE_NAMES) {
+          tooltip_title = SITE_NAMES[participant[property]]
+        }
```
Comment (Elkrival) [inline views/tables/ParticipantsTable/index.jsx]: Nit
```suggestion
        const tooltip_title = participant[property]
        
        if (participant[property] in SITE_NAMES)  tooltip_title = SITE_NAMES[participant[property]]      
```

**Review by Elkrival — APPROVED** [review]
Small formatting nit, non blocking.

File: views/tables/ParticipantsTable/index.jsx (hunk 1 above)
Comment (dheshanm) [reply views/tables/ParticipantsTable/index.jsx]: I see that `tooltip_title`'s value is being reassigned to `const` which suggests immutability will be misleading in this context.

What is the rationale behind changing `tooltip_title` to `const`?

Comment (Elkrival) [reply views/tables/ParticipantsTable/index.jsx]: That's a good question. We want to maintain readable and clean code, reassignments like these can make that a challenge.

Here is another suggestion that makes use of const, this helps in readability and it is pretty clean imo. Also, a reminder that javascript uses `[camelcase](https://developer.mozilla.org/en-US/docs/Glossary/Camel_case)`.

```suggestion
        const tooltipTitle = participant[property] in SITE_NAMES ? SITE_NAMES[participant[property]] : participant[property]
```

Now the variable is immutable, we get the value that we need, we've lessened the number of lines used and reduced mental space keeping track of all of the places that change the value of this variable. 

Let me know your thoughts..?

File: views/tables/ParticipantsTable/index.jsx [hunk 2]
```diff
@@ -135,14 +136,18 @@ const ParticipantsTable = (props) => {
           />
         )
       case 'study':
+        const tooltipTitle = participant[property] in SITE_NAMES ? SITE_NAMES[participant[property]] : participant[property]
```
Comment (mikestone14) [inline views/tables/ParticipantsTable/index.jsx]: I noticed we look at `participant[property]` a number of times in this block. What do you think about extracting that to a variable?

```suggestion
        const participantProperty = participant[property]
        const tooltipTitle = SITE_NAMES[participantProperty] || participantProperty
```

We can use `participantProperty` below in the Typography component as well.

File: views/tables/ParticipantsTable/index.jsx [hunk 3]
```diff
@@ -135,14 +136,19 @@ const ParticipantsTable = (props) => {
...
-          >
-            {participant[property]}
-          </Typography>
+          <Tooltip title={tooltipTitle} placement="right">
+            <Typography
+              component={Link}
+              to={routes.studyDashboard(participant.study)}
+              sx={{ textDecoration: 'none', color: 'text.primary' }}
+            >
+              {participant[property]}
```
Comment (mikestone14) [inline views/tables/ParticipantsTable/index.jsx]: should this be `tooltipTitle`?

Comment (dheshanm) [reply views/tables/ParticipantsTable/index.jsx]: This should have been `participantProperty`. In this context `participantProperty` is the site code (BM) and the `tooltipTitle` is its full name (Birmingham). We are preserving the site code on the participant table and showing the full site name only on hover.

**Review by mikestone14 — APPROVED** [review]

**Pull request merged**